plugins/modules/packaging/os/snap.py

```diff
@@ -185,37 +265,101 @@ def is_snap_enabled(self, snap_name):
         notes = match.group('notes')
         return "disabled" not in notes.split(',')
 
-    def state_present(self):
-        self.vars.meta('classic').set(output=True)
-        self.vars.meta('channel').set(output=True)
-        actionable_snaps = [s for s in self.vars.name if not self.is_snap_installed(s)]
-        if not actionable_snaps:
-            return
+    def process_actionable_snaps(self, actionable_snaps):
         self.changed = True
         self.vars.snaps_installed = actionable_snaps
+
         if self.module.check_mode:
             return
+
         params = ['state', 'classic', 'channel']  # get base cmd parts
         has_one_pkg_params = bool(self.vars.classic) or self.vars.channel != 'stable'
         has_multiple_snaps = len(actionable_snaps) > 1
+
         if has_one_pkg_params and has_multiple_snaps:
             commands = [params + [{'actionable_snaps': [s]}] for s in actionable_snaps]
         else:
             commands = [params + [{'actionable_snaps': actionable_snaps}]]
         self.vars.cmd, rc, out, err = self._run_multiple_commands(commands)
-        if rc == 0:
+
+        if rc != 0:
```

Comment:
Why not simply return if `rc == 0`, and unindent the remainder of this function (as in the original code)?

Reply:
Done.